From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: IBM Codestream linux-2.6.16 october2005, patch 02-14
Patch-Mainline: 2.6.17-rc4

  Description: cio: Enable interrupts on error path.
  Symptom:     Interrupts could stay disabled if error occured in _chp_add().
  Problem:     Use of spin_unlock() instead of spin_unlock_irq().
  Solution:    Use spin_unlock_irq().
  Problem-ID:  23146
  Archs:       s390-31, s390-64

Acked-by: Hannes Reinecke <hare@suse.de>

--- linux-2.5/drivers/s390/cio/chsc.c	12 Apr 2006 11:44:58 -0000	1.131.2.1
+++ linux-2.5/drivers/s390/cio/chsc.c	21 Apr 2006 13:45:45 -0000	1.131.2.2
@@ -639,13 +639,13 @@
 		if (sch->schib.pmcw.chpid[i] == chp->id) {
 			if (stsch(sch->schid, &sch->schib) != 0) {
 				/* Endgame. */
-				spin_unlock(&sch->lock);
+				spin_unlock_irq(&sch->lock);
 				return -ENXIO;
 			}
 			break;
 		}
 	if (i==8) {
-		spin_unlock(&sch->lock);
+		spin_unlock_irq(&sch->lock);
 		return 0;
 	}
 	sch->lpm = ((sch->schib.pmcw.pim &

